prisma/migrations/20250402214855_enable_rls/migration.sql

```diff
@@ -0,0 +1,369 @@
+    -- User-specific policies
+    -- UserProfiles
+    CREATE POLICY "Users can view their own profile" ON "UserProfiles"
+      FOR SELECT USING (auth.uid() = "supId");
+    CREATE POLICY "Users can update their own profile" ON "UserProfiles"
+      FOR UPDATE USING (auth.uid() = "supId");
+
+    -- Wallets
+    CREATE POLICY "Users can view their own wallets" ON "Wallets"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can update their own wallets" ON "Wallets"
+      FOR UPDATE USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own wallets" ON "Wallets"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+    CREATE POLICY "Users can delete their own wallets" ON "Wallets"
+      FOR DELETE USING (auth.uid() = "userId");
+
+    -- WalletActivations
+    CREATE POLICY "Users can view their own wallet activations" ON "WalletActivations"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own wallet activations" ON "WalletActivations"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+    -- WalletRecoveries
+    CREATE POLICY "Users can view their own wallet recoveries" ON "WalletRecoveries"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own wallet recoveries" ON "WalletRecoveries"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+    -- WalletExports
+    CREATE POLICY "Users can view their own wallet exports" ON "WalletExports"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own wallet exports" ON "WalletExports"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+    -- WorkKeyShares
+    CREATE POLICY "Users can view their own work key shares" ON "WorkKeyShares"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own work key shares" ON "WorkKeyShares"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+    CREATE POLICY "Users can update their own work key shares" ON "WorkKeyShares"
+      FOR UPDATE USING (auth.uid() = "userId");
+
+    -- RecoveryKeyShares
+    CREATE POLICY "Users can view their own recovery key shares" ON "RecoveryKeyShares"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own recovery key shares" ON "RecoveryKeyShares"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+
+    -- Challenges
+    CREATE POLICY "Users can view their own challenges" ON "Challenges"
+      FOR SELECT USING (auth.uid() = "userId");
+    CREATE POLICY "Users can insert their own challenges" ON "Challenges"
+      FOR INSERT WITH CHECK (auth.uid() = "userId");
+    CREATE POLICY "Users can update their own challenges" ON "Challenges"
+      FOR UPDATE USING (auth.uid() = "userId");
+    CREATE POLICY "Users can delete their own challenges" ON "Challenges"
+      FOR DELETE USING (auth.uid() = "userId");
+
+    -- DevicesAndLocations
```

Comment:
@matteyu Based on our conversation on Slack, I guess this is what you are having issues with. The policies look goo to me, as `DeviceAndLocation` should never be updated. However, we do use a Prisma `upsert` in `getDeviceAndLocationId()`, but the `update` object is empty, so you could refactor that to use `findFirst` and `insert` (if nothing is found).

Hopefully that resolves the issue without having to add a policy for `UPDATE`.

Once this is fixed, are all other tables/policies working as expected?

Reply:
as it turned out, the issue was actually in the code.  It was how the JWT claims were being set in tandem with using the connection pool